modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/schema/CacheIndexImpl.java

```diff
@@ -190,4 +197,30 @@ public Index queryIndex() {
         // Empty index find predicate.
         return null;
     }
+
+    /** {@inheritDoc} */
+    @Override public boolean isInlineScanPossible(@Nullable ImmutableBitSet requiredColumns) {
+        if (idx == null)
+            return false;
+
+        if (requiredColumns == null)
+            requiredColumns = ImmutableBitSet.range(tbl.descriptor().columnDescriptors().size());
+
+        ImmutableIntList idxKeys = collation.getKeys();
+
+        // All indexed keys should be inlined, all required colummns should be inlined.
+        if (idxKeys.size() < requiredColumns.cardinality() || !ImmutableBitSet.of(idxKeys).contains(requiredColumns))
+            return false;
+
+        List<IndexKeyDefinition> keyDefs = new ArrayList<>(idx.unwrap(InlineIndex.class).indexDefinition()
+            .indexKeyDefinitions().values());
+
+        for (InlineIndexKeyType keyType : InlineIndexKeyTypeRegistry.types(keyDefs, new IndexKeyTypeSettings())) {
+            // Skip variable length keys and java objects (see comments about these limitations in IndexScan class).
```

Comment:
more accurate is to use @see notation here, it would be invalidated in case of class name change.

Reply:
It's not a JavaDoc, it's a comment, `@see` notation doesn't work in comments.